generator/generator.go

```diff
@@ -50,31 +51,42 @@ const (
 )
 
 var (
-	yangPaths              = flag.String("path", "", "Comma separated list of paths to be recursively searched for included modules or submodules within the defined YANG modules.")
-	compressPaths          = flag.Bool("compress_paths", false, "If set to true, the schema's paths are compressed, according to OpenConfig YANG module conventions.")
-	excludeModules         = flag.String("exclude_modules", "", "Comma separated set of module names that should be excluded from code generation this can be used to ensure overlapping namespaces can be ignored.")
-	packageName            = flag.String("package_name", "ocstructs", "The name of the Go package that should be generated.")
-	outputFile             = flag.String("output_file", "", "The file that the generated Go code should be written to.")
-	outputDir              = flag.String("output_dir", "", "The directory that the Go package should be written to.")
-	ignoreCircDeps         = flag.Bool("ignore_circdeps", false, "If set to true, circular dependencies between submodules are ignored.")
+	ocStructsOutputFile     = flag.String("output_file", "", "The file that the generated Go code for manipulating YANG data should be written to. Specify \"-\" for stdout. If both this and output_dir are empty, then schema structs will not be output.")
+	ocStructsOutputDir      = flag.String("output_dir", "", "The directory that the Go package for manipulating YANG data should be written to.")
+	ocPathStructsOutputFile = flag.String("path_structs_output_file", "", "The file that the generated Go code for YANG path construction will be generated. If empty, then path structs are not generated. Specify \"-\" for stdout.")
```

Comment:
It feels a bit strange to be implicit here about whether we're generating path structs or not. Can we rather have an approach whereby we have a flag that controls whether path structs should be output, and then a set of flags that specify a single output file, or an output directory?

I understand that splitting the output files in the base case sounds reasonable - but if we do want to do this, I think we need to think a little more carefully about making this clear to the user.

Reply:
I've added both `generate_structs` and `generate_path_structs` flags to also keep them consistent. However their default values are different to keep backwards compatibility.

`output_dir` is shared across both types of generations.
I've also manually tested the 8 combinations of true/false for these flags, and single vs. multi-file output such that all the tests in the repo continue to pass.